spec.bs

```diff
@@ -18,3 +18,50 @@ Default Biblio Status: current
 Markup Shorthands: markdown yes
 Assume Explicit For: yes
 </pre>
+
+
+<h2 id="advertisements">Advertisements</h2>
+
+An <dfn export>advertisement</dfn>
+is a [=struct=] with the following [=struct/items=]:
+
+* An <dfn for=advertisement>render URL</dfn>. The render URL must have “https” as the scheme.
+* An optional <dfn for=advertisement>metadata</dfn> which is text containing opaque JSON encoded data, passed as an object to auction worklet.
+
+
+<h2 id="advertisement-interest-groups">Advertisement Interest Groups</h2>
+
+An <dfn export>advertisement interest group</dfn>
+is a [=struct=] with the following [=struct/items=]:
+ 
+* An <dfn for="advertisement interest group">expiration</dfn> date.   base::Time expiry;

```

Comment:
I left it as a reminder that I need to find out what type we should use that is equivalent to this base::Time.

Reply:
There is no time or date type either in webidl or in infra, oddly enough.    Just leaving it as "date".

Also, I notice that the idl (in [auction_ad_interest_group.idl](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/ad_auction/auction_ad_interest_group.idl)), does not include the expiration date.  Should it?

Reply:
I think we need to use some type here; I don't think we can just say "date".  I see the HTML spec has a date object, but I dunno if that's what we want.  I gotta imagine some other spec has a concept of a date. Hmm.

auction_ad_interest_group.idl doesn't include expiration because it's part of the navigator.joinAdInterstGroup() API which is defined in navigator_auction.idl

Reply:
We used "a point in time" in https://wicg.github.io/attribution-reporting-api/#attribution-source-source-time. +1 that we don't want to use "date" since it's not just the day part of the time. WebIDL dictionaries and interfaces have to pick a representation for that point in time (often ms-past-the-epoch), but internal structs don't.